components/nodemanager-service/pgdb/nodes.go

```diff
@@ -928,7 +928,9 @@ func (db *DB) QueryManualNodesFields(ctx context.Context, filters []*common.Filt
 			results = append(results, node.Name)
 		case "tags":
 			for _, kv := range node.Tags {
-				results = append(results, kv.Key)
+				if len(kv.Key) > 0 {
```

Comment:
i was getting an extra empty string in the test after the pg call change, so i added this to take care of it. open to suggestions on ways to not end up with the empty string in the first place

Reply:
When you create a new node without any tags a tag with empty key and value is created. The tag can not be deleted. 
![image](https://user-images.githubusercontent.com/1679247/84180208-f9440b80-aa3b-11ea-93bd-c71f4e5ab53d.png)

Reply:
I found one solution. Add the below to line 256.
```
	if len(tags) == 1 && tags[0].Key == "" && tags[0].Value == "" {
		newNode.Tags = []*common.Kv{}
	} else {
		newNode.Tags = tags
	}
```
But I am still looking to see why the empty object is being created.

Reply:
For a node with no tags, we are getting back `[{\"key\": null, \"value\": null}]` from the database.

Reply:
I think I figured it out!
https://stackoverflow.com/questions/24155190/postgresql-left-join-json-agg-ignore-remove-null. 
We use `  COALESCE(json_agg(DISTINCT jsonb_build_object('key', t.key, 'value', t.value)) FILTER (WHERE t.id IS NOT NULL), '[]') AS tags,` to remove the null in the array.

Reply:
nice! i’ll give it a try when i’m done with lunch! thanks!

Reply:
this worked nicely; thanks! i also updated the key to have a wonky string as discussed in standup